stor/obs.py

```diff
@@ -1,7 +1,12 @@
-import cStringIO
 import posixpath
+import sys
 
 from cached_property import cached_property
+from six import BytesIO
+try:
+    from cStringIO import StringIO
+except ImportError:  # pragma: no cover
+    from six import StringIO
```

Comment:
IMO it's better to just use the `io.StringIO` class. It's also implemented in C, and has pretty much the same behavior between Python 2 and Python 3. `cStringIO.StringIO` in python 2 cannot handle non-ASCII unicode correctly, and always returns `str`, even when passed unicode:
```python
>>> import io, cStringIO
>>> # ASCII-encodable unicode
... cStringIO.StringIO(u'asdf').getvalue()
'asdf'
>>> io.StringIO(u'asdf').getvalue()
u'asdf'
>>> 
>>> # non-ASCII-encodable unicode
... cStringIO.StringIO(u'💣').getvalue()
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
UnicodeEncodeError: 'ascii' codec can't encode characters in position 0-1: ordinal not in range(128)
>>> io.StringIO(u'💣').getvalue()
u'\U0001f4a3'
>>> 
>>> # Bytes
... cStringIO.StringIO('asdf').getvalue()
'asdf'
>>> io.StringIO('asdf').getvalue()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: initial_value must be unicode or None, not str
```
```